extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -467,7 +465,30 @@ public function getCuratedContentSection() {
 			throw new NotFoundApiException( 'No members' );
 		}
 
-		$this->response->setVal( 'items', $data[ 'items' ] );
+		$this->response->setVal( 'items', $data['items'] );
+	}
+
+	public function getMainPageDetailsAndAdsContext() {
+		$mainPageTitle = Title::newMainPage();
+		$mainPageArticleID = $mainPageTitle->getArticleID();
+		$article = Article::newFromID( $mainPageArticleID );
+
+		$articleDetails = $this->getArticleDetails( $article );
+
+		if ( empty( $articleDetails ) ) {
+			throw new WikiaException( 'Article details empty', 500 );
```

Comment:
you are asking API for these values.
If API can't provide value I think it should throw.
How this exception is going to be handled is another thing.

Exception is an unambiguous way of saying something is wrong.